src/server/api/routers/blogPost.ts

```diff
@@ -47,4 +47,30 @@ export const blogPostRouter = createTRPCRouter({
         nextCursor,
       };
     }),
+  update: protectedProcedure.input(BlogPostUpdateOneSchema).mutation(async ({ ctx, input }) => {
+      const user = await ctx.prisma.user.findUniqueOrThrow({
+        where: {
+          id: ctx.session.user.id
+        }
+      });
+
+      const blogPost = await ctx.prisma.blogPost.findUnique({
+        where: {
+          id: input.where.id
+        }
+      });
+
+      if (!["CONTRIBUTOR", "ADMIN"].includes(user.role)) {
+        throw new Error("Your account role is not permitted to update blog posts");
+      }
+
+      if (blogPost !== null) {
```

Comment:
^ same with previous, and if we use findUniqueOrThrow, there is no need for this conditional